src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/EnumeratingWhitelist.java

```diff
@@ -218,6 +218,7 @@ private static boolean is(String thisIdentifier, String identifier) {
         }
         abstract boolean exists() throws Exception;
         final Class<?> type(String name) throws Exception {
+            // TODO this should be more strict: binary name is required for nested classes
             return ClassUtils.getClass(name);
```

Comment:
Maybe we could add an assertion here so that any issues with the static whitelists will cause the plugin's tests to fail?

```suggestion
            Class<?> clazz = ClassUtils.getClass(name);
            assert clazz.getName().equals(name); // The whitelists require binary names, but ClassUtils allows various syntaxes.
            return clazz;
```

Reply:
Well that would basically be the same as
```suggestion
            return Class.forName(name);
```
which will not work (AFAIK) because `name` is sometimes something like `int` or `Object[]`. We want to enforce use of binary name on the actual class name portion of the string, while still processing special syntaxes that `ClassUtils` supports. I spent a few minutes thinking about it and did not come up with a simple fix so I moved on.